lib/harvestdor/indexer.rb

```diff
@@ -63,11 +63,16 @@ def harvest_and_index(each_options = { in_threads: 4 })
     end
 
     def resources
-      druids.map do |x|
-        Harvestdor::Indexer::Resource.new(self, x)
-      end.map do |x|
-        [x, (x.items if x.collection?)]
-      end.flatten.uniq.compact
+      return to_enum(:resources) unless block_given?
+
+      druids.each do |x|
+        resource = Harvestdor::Indexer::Resource.new(self, x)
+        yield resource
+
+        resource.items.each do |member|
+          yield member
+        end if resource.collection?
+      end
```

Comment:
oh yeah:  and a note about  the possibility that ids might be repeated in the returned enumerable  (a @return comment indicating the type of thingy returned wouldn't go amiss, AFAIC) but Solr can handle that ok, and it beats persisting the entire list for each object indexed ??